Azure Billing Metrics UX and docs update

zmoog opened the pull request:
<!-- Type of change
Please label this PR with one of the following labels, depending on the scope of your change:
- Bug
- Enhancement
- Breaking change
- Deprecation
-->

## What does this PR do?

<!-- Mandatory
Explain here the changes you made on the PR.
-->

Update the integration UX and documentation to enhance:

- How to create and configure the required App Registration.
- How to use the scopes (subscription, department, and billing account).

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [ ] ~~I have verified that all data streams collect metrics or logs.~~
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

<!-- Recommended
## Author's Checklist
Add a checklist of things that are required to be reviewed in order to have the PR approved
- [ ]
-->

## How to test this PR locally

<!-- Recommended
Explain here how this PR will be tested by the reviewer: commands, dependencies, steps, etc.
-->

<!-- Recommended
## Related issues
Link related issues below. Insert the issue link or reference after the word "Closes" if merging this should automatically close it.

- Closes #123
- Relates #123
- Requires #123
- Supersedes #123
-->
- Closes #3279

## Screenshots

<!-- Optional
Add here screenshots presenting:
- Kibana UI forms presenting configuration options exposed by the integration
- dashboards with collected metrics or logs
-->

### Settings

![CleanShot 2022-09-22 at 22 43 20@2x](https://user-images.githubusercontent.com/25941/191847307-0e0703e0-32ae-48fd-91e9-1a4a6f087863.png)

### Documentation

![CleanShot 2022-09-22 at 22 26 17@2x](https://user-images.githubusercontent.com/25941/191844476-d31bd69f-e209-4820-9315-7549965e345f.png)

elasticmachine commented:
## :broken_heart: Build Failed


<!-- BUILD BADGES-->
> _the below badges are clickable and redirect to their specific view in the CI or DOCS_
[![Pipeline View](https://img.shields.io/badge/pipeline-pipeline%20-green)](https://fleet-ci.elastic.co/blue/organizations/jenkins/Ingest-manager%2Fintegrations%2FPR-4219/detail/PR-4219/2//pipeline) [![Test View](https://img.shields.io/badge/test-test-green)](https://fleet-ci.elastic.co/blue/organizations/jenkins/Ingest-manager%2Fintegrations%2FPR-4219/detail/PR-4219/2//tests) [![Changes](https://img.shields.io/badge/changes-changes-green)](https://fleet-ci.elastic.co/blue/organizations/jenkins/Ingest-manager%2Fintegrations%2FPR-4219/detail/PR-4219/2//changes) [![Artifacts](https://img.shields.io/badge/artifacts-artifacts-yellow)](https://fleet-ci.elastic.co/blue/organizations/jenkins/Ingest-manager%2Fintegrations%2FPR-4219/detail/PR-4219/2//artifacts) [![preview](https://img.shields.io/badge/docs-preview-yellowgreen)](http://integrations_4219.docs-preview.app.elstc.co/diff)  [![preview](https://img.shields.io/badge/elastic-observability-blue)](https://ci-stats.elastic.co/app/apm/services/fleet-ci/transactions/view?rangeFrom=2022-09-14T18:27:46.976Z&rangeTo=2022-09-14T18:47:46.976Z&transactionName=BUILD+Ingest-manager%2Fintegrations%2FPR-%7Bnumber%7D&transactionType=job&latencyAggregationType=avg&traceId=54af91dee046da7d0f00be8a2054a75d&transactionId=68ce0cb7e794a203)

<!-- BUILD SUMMARY-->
<details><summary>Expand to view the summary</summary>
<p>

#### Build stats




* Start Time: 2022-09-14T18:37:46.976+0000


* Duration: 9 min 46 sec





</p>
</details>

<!-- TEST RESULTS IF ANY-->


<!-- STEPS ERRORS IF ANY -->














### Steps errors [![2](https://img.shields.io/badge/2%20-red)](https://fleet-ci.elastic.co/blue/organizations/jenkins/Ingest-manager%2Fintegrations%2FPR-4219/detail/PR-4219/2//pipeline)
  <details><summary>Expand to view the steps failures</summary>
  <p>


  
  
  
  
##### `Check integration: azure_billing`
<ul>
<li>Took 0 min 0 sec . View more details <a href="https://fleet-ci.elastic.co//blue/rest/organizations/jenkins/pipelines/Ingest-manager/pipelines/integrations/pipelines/PR-4219/runs/2/steps/920/log/?start=0">here</a></li>
<li>Description: <code>../../build/elastic-package check -v</code></l1>
</ul>
  
  
  
  
##### `Google Storage Download`
<ul>
<li>Took 0 min 0 sec . View more details <a href="https://fleet-ci.elastic.co//blue/rest/organizations/jenkins/pipelines/Ingest-manager/pipelines/integrations/pipelines/PR-4219/runs/2/steps/957/log/?start=0">here</a></li>

</ul>
  
  </p>
  </details>



## :robot: GitHub comments

To re-run your PR in the CI, just comment with:



- `/test` : Re-trigger the build.




<!--COMMENT_GENERATED_WITH_ID_comment.id-->

elasticmachine commented:
### :globe_with_meridians: Coverage report

Name | Metrics % (`covered/total`) | Diff
 --- | --- | ---
Packages | 100.0% (`0/0`) | :green_heart: 
Files | 100.0% (`0/0`) | :green_heart: 2.674
Classes | 100.0% (`0/0`) | :green_heart: 2.674
Methods | 66.667% (`2/3`) | :-1: -23.116
Lines | 100.0% (`0/0`) | :green_heart: 8.558
Conditionals | 100.0% (`0/0`) | :green_heart: 

<!--COMMENT_GENERATED_WITH_ID_coverage-->

colleenmcginnis reviewed (commented):
👋  @zmoog thanks for improving the docs!

I took a first look at the changes, and I have some concerns about the order of the content. A few months ago we developed [documentation guidelines](https://github.com/elastic/integrations/blob/main/docs/documentation_guidelines.md#documentation-guidelines) for integrations. Part of the intention behind the guidelines was to gradually introduce more complex information to the reader as they are deciding whether or not to use the integration, setting it up, and customizing it for their use case.

For example, here's the thought process a reader might go through when visiting the integration docs:
1. The first thing they may want to know is what is this particular Azure integration for? (That should be covered in the [Overview](https://github.com/elastic/integrations/blob/main/docs/documentation_guidelines.md#overview) section.)
2. They probably have in mind what kind of data they want from Azure. We should describe that in a bit more detail next (in the [Data streams](https://github.com/elastic/integrations/blob/main/docs/documentation_guidelines.md#data-streams) section) so they can be confident that they are looking at the _right_ Azure integration before getting into the details of implementation. 
3. Then, once they are sold on this particular integration, we should be upfront about any requirements or limitations of the integration so they can again be confident that this integration will work with their use case (for example, knowing it will work with specific versions of a product). (This should be covered in the [Requirements](https://github.com/elastic/integrations/blob/main/docs/documentation_guidelines.md#requirements) section.)
4. Then, once they are sure this is the right integration and that there are no deal breakers for using this integration with their systems, they are ready to set up the integration both from the Elastic side and the Azure side. (This should be covered in the [Setup](https://github.com/elastic/integrations/blob/main/docs/documentation_guidelines.md#setup) section.)
5. Finally, include additional information that is not necessarily required for all users to know, but may be very helpful to some. (This includes [Reference](https://github.com/elastic/integrations/blob/main/docs/documentation_guidelines.md#reference) content and [Troubleshooting](https://github.com/elastic/integrations/blob/main/docs/documentation_guidelines.md#troubleshooting) tips.)

If there's additional information to include that we think would be helpful to the reader, but does not fit into any of these categories (maybe "How it works" or "Cost"), we should talk about whether that should be added to the guidelines so other integrations can also benefit from those additions! 

Let me know if you have any questions or concerns about reordering this content to align with the documentation guidelines (or if you would like me to make some changes to the order and commit them to this PR).

zmoog commented:
> Let me know if you have any questions or concerns about reordering this content to align with the documentation guidelines

Hey @colleenmcginnis, I am more than happy to align this doc to the guidelines! Consistency is king. I was aware it existed from you mentioning it in other PRs, but I didn't recall it when I started! 

I will follow up with other comments.

Thank you so much for spending some time on the review!

zmoog commented:
@colleenmcginnis, can I use the AWS docs as a reference?

colleenmcginnis commented:
Yes! AWS would be a good reference. Though AWS has the central [AWS integration](https://docs.elastic.co/integrations/aws) and I don't think that exists for Azure, right? 

We might be able to create a kind of "landing page" that is not associated with a specific integration, but contains an overview of Azure-related integrations if it would be helpful.

zmoog commented:
Hey, @colleenmcginnis, thanks again for clarifying the purpose of the guideline; I think it's a great thing.

> If there's additional information to include that we think would be helpful to the reade, but does not fit into any of these categories (maybe "How it works" or "Cost"), we should talk about whether that should be added to the guidelines so other integrations can also benefit from those additions!

I believe that both "How it works" and "Cost" sections do not deserve a spot at the top level of the guidelines: overview, data stream, requirements, setup, and reference work great.

In this new revision, I am experimenting with this approach:

- Moved  "How it works" as a sub-section in Setup, so users can have an idea of what we are setting up at a glance. It's more of a visual overview of the result.
- Moved "Cost" as a sub-section of Requirement. I'm not 100% sure, but it may be a good fit in the section that should summarize the requirements and limitations. Like you are _required_ to spend a little money to pull this data from Azure.

I also moved the "Settings" section at the end of the "Setup". The idea is to provide guidance about configuring the integration's setting page. Another approach would be to shrink this section by moving details in the field description on the setting page and leaving just a few field reference information, like the possible value for the endpoints ("Resource Manager Endpoint" and "Active Directory Endpoint").

@colleenmcginnis I'd love to know what you think. I'm open to suggestions and willing to move stuff around until we are both happy with the result.

zmoog commented on packages/azure_billing/docs/README.md:
```diff
@@ -1,68 +1,173 @@
+# Azure Billing Metrics Integration
 
-The [Azure Billing](https://docs.microsoft.com/en-us/azure/cost-management-billing/) Integration allows users to retrieve usage details and forecast information of the subscription configured.
+The Azure Billing Metrics integration allows users to monitor their Azure spending to optimize resource use.
```
You're right, it feels more natural and easy to read.

zmoog commented on packages/azure_billing/docs/README.md:
```diff
@@ -1,68 +1,173 @@
-For a more detailed walk-through, see: [Using Azure PowerShell to create a service principal to access resources](https://docs.microsoft.com/en-us/powershell/azure/create-azure-service-principal-azureps?view=azps-2.7.0).
-It is also possible to create a service principal via the [Azure portal](https://docs.microsoft.com/en-us/azure/active-directory/develop/howto-create-service-principal-portal).
-Users will have to make sure the roles assigned to the application contain at least reading permissions to the monitor data. See: [Role-based access control](https://docs.microsoft.com/en-us/azure/role-based-access-control/built-in-roles).
+## How it works
```
I'm experimenting by moving it as a sub-section in Setup. Let me know what you think!

zmoog commented on packages/azure_billing/docs/README.md:
```diff
@@ -1,68 +1,173 @@
+The integration retrieves two kinds of data:
 
+* [Usage details](https://docs.microsoft.com/en-us/azure/cost-management-billing/manage/consumption-api-overview#usage-details-api), from the Consumption API.
+* [Forecast](https://docs.microsoft.com/en-us/rest/api/cost-management/forecast) information from the Cost Management API.
```
Yep, I agree. Moved to the Overview section!

zmoog commented on packages/azure_billing/docs/README.md:
```diff
@@ -1,68 +1,173 @@
+* [Quickstart: Register an application with the Microsoft identity platform](https://docs.microsoft.com/en-us/azure/active-directory/develop/quickstart-register-app) 
+* [Use the portal to create an Azure AD application and service principal that can access resources](https://docs.microsoft.com/en-us/azure/active-directory/develop/howto-create-service-principal-portal)
+
+This section of the integration docs is an adaptation for the Elastic Agent of these two guides.
```
Yeah, I agree! Moved.

tommyers-elastic commented on packages/azure_billing/_dev/build/docs/README.md:
```diff
@@ -1,76 +1,199 @@
+## Requirements
 
-The azure credentials keys can be used if configured `AZURE_CLIENT_ID`, `AZURE_CLIENT_SECRET`, `AZURE_TENANT_ID`, `AZURE_SUBSCRIPTION_ID`
+You need Elasticsearch to store and search your data and Kibana to visualize and manage it. You can use our hosted Elasticsearch Service on Elastic Cloud, which is recommended, or self-manage the Elastic Stack on your hardware.
```
tiny nit: what about our OOTB self-managed cloud offerings like https://azuremarketplace.microsoft.com/en-us/marketplace/apps/elastic.elasticsearch?tab=Overview

colleenmcginnis commented on packages/azure_billing/_dev/build/docs/README.md:
```diff
@@ -1,76 +1,199 @@
+
+#### Assign Role
+
+1. In the [Azure Portal](https://portal.azure.com/), search for and select Subscriptions.
+1. Select the subscription to assign the application.
+1. Select Access control (IAM).
+1. Select Add > Add role assignment to open the Add role assignment page.
+1. In the Role tab, search and select the role "Billing Reader".
+1. Select the Next button to move to the Members tab.
+1. Select Assign access to-> User, group, or service principal, and select Select members. This page does not display Azure AD applications in the available options by default.
+1. To find your application, search by name (for example, "elastic-agent") and select it from the list.
+1. Click the Select button.
+1. Then click the Review + assign button.
```
```suggestion
1. In the [Azure Portal](https://portal.azure.com/), search for and select **Subscriptions**.
1. Select the subscription to assign the application.
1. Select **Access control (IAM)**.
1. Select **Add** > **Add role assignment** to open the _Add role assignment page_.
1. In the **Role** tab, search and select the role **Billing Reader**.
1. Select the **Next** button to move to the **Members** tab.
1. Select **Assign access to** > **User, group, or service principal**, and select **Select members**. This page does not display Azure AD applications in the available options by default.
1. To find your application, search by name (for example, "elastic-agent") and select it from the list.
1. Click the **Select** button.
1. Then click the **Review + assign** button.
```

colleenmcginnis commented on packages/azure_billing/_dev/build/docs/README.md:
```diff
@@ -1,76 +1,199 @@
+1. Click the Select button.
+1. Then click the Review + assign button.
+
+Take note of the following values. We will use them in the Settings section later:
+
+* `Subscription ID`: use the content of the "Subscription ID" you selected.
+* `Tenant ID`: use the "Tenant ID" from the Azure Active Directory you use.
+
+Your App Registration is now ready for the Elastic Agent.
```
```suggestion
Take note of the following values, which you will use later when specifying settings.

* `Subscription ID`: use the content of the "Subscription ID" you selected.
* `Tenant ID`: use the "Tenant ID" from the Azure Active Directory you use.

Your App Registration is now ready for the Elastic Agent.
```

colleenmcginnis commented on packages/azure_billing/_dev/build/docs/README.md:
```diff
@@ -1,76 +1,199 @@
+
+### Settings
+
+#### Main Options
```
```suggestion
#### Main options
```

zmoog replied:
Do you have some resources I can read about the correct capitalization? Thanks!

colleenmcginnis commented on packages/azure_billing/_dev/build/docs/README.md:
```diff
@@ -1,76 +1,199 @@
+
+#### Main Options
+
+The settings' main section contains all the options to access the Azure APIs and collect the billing data. You will now use all the notes you collected in the previous sections.
+
+`Client ID` ::
+_string_
+The unique identifier of the App Registration (sometimes referred to as Application ID).
+
+`Client Secret` ::
+_string_
+The client secret for authentication.
+
+`Subscription ID` ::
+_string_
+The unique identifier for the Azure subscription. You can provide just one subscription ID. The Agent uses this ID to access Azure APIs. The Agent also uses this ID as the default scope for billing information: see the "Scope" section for more details about how to collect data for more than one subscription.
+
+`Tenant ID` :: The unique identifier of the Azure Active Directory's Tenant ID.
```
Mostly just formatting updates:

```suggestion
The settings' main section contains all the options needed to access the Azure APIs and collect the billing data. You will now use all the values from [App registration](#app-registration) including:

`Client ID` _string_
: The unique identifier of the App Registration (sometimes referred to as Application ID).

`Client Secret` _string_
: The client secret for authentication.

`Subscription ID` _string_
: The unique identifier for the Azure subscription. You can provide just one subscription ID. The Agent uses this ID to access Azure APIs. The Agent also uses this ID as the default scope for billing information: see the "Scope" section for more details about how to collect data for more than one subscription.

`Tenant ID` _string_
: The unique identifier of the Azure Active Directory's Tenant ID.
```

| Before | After |
|---|---|
| <img width="881" alt="Screen Shot 2022-09-20 at 1 17 02 PM" src="https://user-images.githubusercontent.com/10479155/191334080-dd6b74b3-9a3c-4064-8ed8-ab63d38bb9e4.png"> | <img width="871" alt="Screen Shot 2022-09-20 at 1 16 42 PM" src="https://user-images.githubusercontent.com/10479155/191334123-9a721d96-d2bd-4995-b0b6-5514f11d5c50.png"> |

zmoog replied:
Whoa, it's much better now, thanks!

colleenmcginnis commented on packages/azure_billing/_dev/build/docs/README.md:
```diff
@@ -1,76 +1,199 @@
+
+#### Advanced Options
+
+In addition, there are two additional advanced options:
```
```suggestion
There are two additional advanced options:
```

colleenmcginnis commented on packages/azure_billing/_dev/build/docs/README.md:
```diff
@@ -1,76 +1,199 @@
+In addition, there are two additional advanced options:
 
 `Resource Manager Endpoint` ::
 _string_
-Optional, by default the azure public environment will be used, to override, users can provide a specific resource manager endpoint in order to use a different azure environment.
+Optional. By default, the integration uses the Azure public environment. To override, users can provide a specific resource manager endpoint to use a different Azure environment.
```
I can't suggest an edit here, but it should look like the reformatted definition lists above:

```markdown
`Resource Manager Endpoint`  _string_
: Optional. By default, the integration uses the Azure public environment. To override, users can provide a specific resource manager endpoint to use a different Azure environment.
```

colleenmcginnis commented on packages/azure_billing/_dev/build/docs/README.md:
```diff
@@ -1,76 +1,199 @@
+
 Examples:
+
 * https://management.chinacloudapi.cn for azure ChinaCloud
 * https://management.microsoftazure.de for azure GermanCloud
 * https://management.azure.com for azure PublicCloud
 * https://management.usgovcloudapi.net for azure USGovernmentCloud
```
Should "azure" be capitalized, too? 

```suggestion
* `https://management.chinacloudapi.cn` for azure ChinaCloud
* `https://management.microsoftazure.de` for azure GermanCloud
* `https://management.azure.com` for azure PublicCloud
* `https://management.usgovcloudapi.net` for azure USGovernmentCloud
```

zmoog replied:
Yeah, I think so. I will capitalize it.

colleenmcginnis commented on packages/azure_billing/_dev/build/docs/README.md:
```diff
@@ -1,76 +1,199 @@
 * https://management.usgovcloudapi.net for azure USGovernmentCloud
 
 `Active Directory Endpoint` ::
 _string_
-Optional, by default the associated active directory endpoint to the resource manager endpoint will be used, to override, users can provide a specific active directory endpoint in order to use a different azure environment.
+Optional. By default, the integration uses the associated Active Directory Endpoint. To override, users can provide a specific active directory endpoint to use a different Azure environment.
```
Same here: 

```markdown
`Active Directory Endpoint` _string_
: Optional. By default, the integration uses the associated Active Directory Endpoint. To override, users can provide a specific active directory endpoint to use a different Azure environment.
```

colleenmcginnis commented on packages/azure_billing/_dev/build/docs/README.md:
```diff
@@ -1,76 +1,199 @@
+
 Examples:
+
 * https://login.microsoftonline.com for azure ChinaCloud
 * https://login.microsoftonline.us for azure GermanCloud
 * https://login.chinacloudapi.cn for azure PublicCloud
 * https://login.microsoftonline.de for azure USGovernmentCloud
```
```suggestion
* `https://login.microsoftonline.com` for azure ChinaCloud
* `https://login.microsoftonline.us` for azure GermanCloud
* `https://login.chinacloudapi.cn` for azure PublicCloud
* `https://login.microsoftonline.de` for azure USGovernmentCloud
```

colleenmcginnis commented on packages/azure_billing/_dev/build/docs/README.md:
```diff
@@ -1,76 +1,199 @@
-The integration contains the following data streams:
-
-### billing
+#### Data Stream Options
```
```suggestion
#### Data stream options
```

colleenmcginnis commented on packages/azure_billing/_dev/build/docs/README.md:
```diff
@@ -1,76 +1,199 @@
-### billing
+#### Data Stream Options
 
+The data stream has some additional options about scope and period. Please read the "Scope" section to learn more about the scope.
```
```suggestion
The data stream has some additional options about scope and period. To learn more about the scope, read the [Scope](#scope) section.
```

colleenmcginnis commented on packages/azure_billing/_dev/build/docs/README.md:
```diff
@@ -1,76 +1,199 @@
 
+The data stream has some additional options about scope and period. Please read the "Scope" section to learn more about the scope.
 
+`Billing Scope Department ID`:: (_string_) Retrieve data based on the department ID.
```
```suggestion
`Billing Scope Department ID` _string_
: Retrieve data based on the department ID.
```

colleenmcginnis commented on packages/azure_billing/_dev/build/docs/README.md:
```diff
@@ -1,76 +1,199 @@
+`Billing Scope Department ID`:: (_string_) Retrieve data based on the department ID.
 
-#### Configuration options
+`Billing Scope Account ID`:: (_string_) Retrieve data based on the billing account ID. The billing account ID is available on the [Azure Portal](https://portal.azure.com/) at Cost Management + Billing, select a billing scope of the type "billing account", then Setting > Properties > ID.
```
```suggestion
`Billing Scope Account ID`  _string_
: Retrieve data based on the billing account ID. The billing account ID is available on the [Azure Portal](https://portal.azure.com/) at **Cost Management + Billing**, select a billing scope of the type "billing account", then **Setting** > **Properties** > **ID**.
```

colleenmcginnis commented on packages/azure_billing/_dev/build/docs/README.md:
```diff
@@ -1,76 +1,199 @@
-#### Configuration options
+`Billing Scope Account ID`:: (_string_) Retrieve data based on the billing account ID. The billing account ID is available on the [Azure Portal](https://portal.azure.com/) at Cost Management + Billing, select a billing scope of the type "billing account", then Setting > Properties > ID.
 
 `Period`:: (_string_) The time interval to use when retrieving metric values.
```
```suggestion
`Period` _string_
: The time interval to use when retrieving metric values.
```

colleenmcginnis commented on packages/azure_billing/_dev/build/docs/README.md:
```diff
@@ -1,76 +1,199 @@
 
+The integration uses the Subscription ID as the default scope for the billing data.
 
+To change the scope, expand the data stream section named "Collect Azure Billing metrics" in the integration settings and set one of the two available options (if you set both, the billing account scope take precedence over the department):
```
```suggestion
To change the scope, expand the data stream section named **Collect Azure Billing metrics** in the integration settings and set one of the two available options (if you set both, the billing account scope take precedence over the department):
```

colleenmcginnis commented on packages/azure_billing/_dev/build/docs/README.md:
```diff
@@ -1,76 +1,199 @@
 
+To change the scope, expand the data stream section named "Collect Azure Billing metrics" in the integration settings and set one of the two available options (if you set both, the billing account scope take precedence over the department):
 
+* `Billing Scope Department ID` :: Collect user details and forecast data for the given deparment ID.
+* `Billing Scope Account ID` :: Collect user details and forecast data for the given billing account ID.
```
```suggestion
* `Billing Scope Department ID` : Collect user details and forecast data for the given department ID.
* `Billing Scope Account ID` : Collect user details and forecast data for the given billing account ID.
```

colleenmcginnis commented on packages/azure_billing/_dev/build/docs/README.md:
```diff
@@ -1,76 +1,199 @@
+* [Quickstart: Register an application with the Microsoft identity platform](https://docs.microsoft.com/en-us/azure/active-directory/develop/quickstart-register-app) 
+* [Use the portal to create an Azure AD application and service principal that can access resources](https://docs.microsoft.com/en-us/azure/active-directory/develop/howto-create-service-principal-portal)
+
+### Settings
+
```
```suggestion
### Settings

Add the Azure Billing Metrics integration in Kibana and specify settings.

If you're new to integrations, you can find  step-by-step instructions on how to set up an integration in the {{ url "getting-started-observability" "Getting started" }} guide.

```

colleenmcginnis reviewed (commented):
I like how this content flows now. Thanks for aligning it with the documentation guidelines!

I left lots of comments/suggestions below, but most are relatively small. Let me know if you have any questions or concerns!

zmoog commented on packages/azure_billing/_dev/build/docs/README.md:
```diff
@@ -1,76 +1,199 @@
+The integration uses [Azure Consumption API](https://docs.microsoft.com/en-us/azure/cost-management-billing/manage/consumption-api-overview#usage-details-api) to collect usage details and leverages [Azure Cost Management API](https://docs.microsoft.com/en-us/rest/api/cost-management/forecast) to bring forecast data.
 
-Required credentials for the `azure_billing` integration:
+Use the Azure Billing Metrics integration to collect detailed resource usage and forecast expense for the coming weeks. For example, if you want to know which resources cost you more, you could check the top resources pie chart. Then you can visualize the prediction for the coming weeks by looking at the forecast chart.
```
You're right, it's a donut 🍩 indeed! 🤦

zmoog commented on packages/azure_billing/_dev/build/docs/README.md:
```diff
@@ -1,76 +1,199 @@
+## Data streams
 
-`Client Secret`:: The client/application secret/key
+### Azure Billing Metrics
```
Yeah, I wasn't sure. I think you made the right call.

zmoog commented on packages/azure_billing/_dev/build/docs/README.md:
```diff
@@ -1,76 +1,199 @@
+## Requirements
 
-The azure credentials keys can be used if configured `AZURE_CLIENT_ID`, `AZURE_CLIENT_SECRET`, `AZURE_TENANT_ID`, `AZURE_SUBSCRIPTION_ID`
+You need Elasticsearch to store and search your data and Kibana to visualize and manage it. You can use our hosted Elasticsearch Service on Elastic Cloud, which is recommended, or self-manage the Elastic Stack on your hardware.
+
+You need to set up an Azure App Registration to allow the Agent to access the Azure APIs. See more details in the Setup section.
+
+The App Registration requires the Billing Reader role to access the billing information for the subscription, department or billing account. See more details in the Setup section.
+
+Azure Billing Metrics integration queries are charged based on the number of standard API calls. One integration makes two calls every 24 hours in the standard configuration.
```
"Payment method" is what I was looking for! It wasn't hard, but now it seems obvious in retrospect.

zmoog commented on packages/azure_billing/_dev/build/docs/README.md:
```diff
@@ -1,76 +1,199 @@
+
+Follow these steps to create the app registration:
+
+1. Sign in to the [Azure Portal](https://portal.azure.com/).
+2. Search for and select Azure Active Directory.
+3. Under Manage, select App registrations > New registration.
+4. Enter a display Name for your application. Possible names are "elastic-agent."
+5. Specify who can use the application.
+6. Don't enter anything for Redirect URI. It's optional; the Agent doesn't use it.
+7. Select Register to complete the initial app registration.
```
Cleaner and smoother! 💯

zmoog commented on packages/azure_billing/_dev/build/docs/README.md:
```diff
@@ -1,76 +1,199 @@
+6. Don't enter anything for Redirect URI. It's optional; the Agent doesn't use it.
+7. Select Register to complete the initial app registration.
+
+Take note of the following value. We will use it in the Settings section later:
+
+`Client ID`: use the content of the "Application (client) ID."
+
+You now have a new App Registration ready for the next steps.
```
This is not a big deal, but we need to copy the page element labeled as "Application (client) ID" and take note of its value as "Client ID".

![CleanShot 2022-09-20 at 22 28 32@2x](https://user-images.githubusercontent.com/25941/191358849-6957a589-87ab-4854-8600-3014a09bd2e3.png)


The first time I created an App Registration, I wasn't sure this was the right field, so I highlighted the mapping between the two explicitly in the docs. 

Do you think it is not necessary?

colleenmcginnis replied:
Oh interesting. What do you think about something like this?  

```suggestion
Take note of the **Application (client) ID**, which you will use later when specifying the **Client ID** in the integration settings.
```

zmoog commented on packages/azure_billing/_dev/build/docs/README.md:
```diff
@@ -1,76 +1,199 @@
+
+This integration uses Client Secrets to prove its identity.
+
+1. In the [Azure Portal](https://portal.azure.com/), select the application we created in the previous "Register a new App" section.
+1. Select Certificates & secrets > Client secrets > New client secret.
+1. Add a description (for example, "Elastic Agent client secrets").
+1. Select an expiration for the secret or specify a custom lifetime.
+1. Select Add.
```
Hey @colleenmcginnis, sorry for messing up with pronouns you/we/whatever all over the doc. So "you" should be our default choice in this kind of document?

zmoog commented on packages/azure_billing/_dev/build/docs/README.md:
```diff
@@ -1,76 +1,199 @@
+1. Select an expiration for the secret or specify a custom lifetime.
+1. Select Add.
+
+Take note of the following value. We will use it in the Settings section later:
+
+`Client Secret`: use the content of the "Value" field.
+
+This secret value is never displayed again after you leave this page. Record the secret's value in a safe place. You will need it on the integration's settings page.
```
This is another place where it's not 100% obvious at glance what value should be copied:

![CleanShot 2022-09-20 at 22 39 03@2x](https://user-images.githubusercontent.com/25941/191360202-6598a060-5082-4134-92b9-12e482041604.png)

The right page element to take note of as "Client Secret" is the one labeled as "Value". 

Since we're looking for a "Client Secret", I found the page element labeled "Secret ID" somewhat confusing, but maybe it's just me.

colleenmcginnis replied:
Similar to the comment above, what about this?

```suggestion
Take note of the content in the **Value** column in the **Client secrets** table, which you will use later when specifying a **Client Secret** in the integration settings. **This secret value is never displayed again after you leave this page.** Record the secret's value in a safe place.
```

zmoog commented:
@colleenmcginnis, do you think this PR needs more work on the docs?

colleenmcginnis reviewed (approved):
A few minor comments below, but after those are addressed this looks good from my side!

zmoog commented on packages/azure_billing/_dev/build/docs/README.md:
```diff
@@ -1,76 +1,203 @@
-* https://management.microsoftazure.de for azure GermanCloud
-* https://management.azure.com for azure PublicCloud
-* https://management.usgovcloudapi.net for azure USGovernmentCloud
+Optional. By default, the integration uses the Azure public environment. To override, users can provide a specific resource manager endpoint to use a different Azure environment.
```
Fixed.
